my_instance.tf

```diff
@@ -60,39 +34,41 @@ resource "aws_security_group" "ec2_sg" {
 
 }
 
+#Create AMI instance
+data "aws_ami" "ubuntu_latest" {
+  owners      = ["099720109477"]
+  most_recent = true
+
+  filter {
+    name   = "name"
+    values = ["ubuntu/images/hvm-ssd/ubuntu-jammy-22.04-amd64-server-*"]
+  }
+}
 
 #Create Ec2 instance
 resource "aws_instance" "intro" {
-  ami                    = "ami-053b0d53c279acc90"
-  instance_type          = "t2.micro"
-  availability_zone      = "us-east-1a"
-  key_name               = aws_key_pair.terra-key.key_name
-  vpc_security_group_ids = [aws_security_group.ec2_sg.name]
-  tags                   = merge(var.common_tags,{Name = "Ec2_instance"}) 
-
+  ami                  = data.aws_ami.ubuntu_latest.id
+  instance_type        = "t2.micro"
+  availability_zone    = var.access_zones
+  iam_instance_profile = aws_iam_instance_profile.s_m_read_write.name
+  user_data            = <<EOF
+sudo apt update
+sudo apt upgrade
 
-  /*provisioner "file" {
-    source      = "web.sh"
-    destination = "/tmp/web.sh"
-  }*/
+sudo apt install unzip -y
+sudo curl "https://awscli.amazonaws.com/awscli-exe-linux-x86_64.zip" -o "awscliv2.zip"
 
+sudo unzip awscliv2.zip
+sudo ./aws/install
 
-  /*provisioner "remote-exec" {
-    inline = [
-      "chmod u+x /tmp/web.sh",
-      "sudo /tmp/web.sh"
 
-    ]
-  }
-
-
-  connection {
-    user        = var.USER
-    private_key = file("terra-key")
-    host        = self.public_ip
-  }*/
+aws secretsmanager get-secret-value --secret-id ${aws_secretsmanager_secret.my-test-secret2.arn}
```

Comment:
you got value of secret but you did not write it into a .env file :)

Reply:
fixed